cmd/guacone/cmd/patch.go

```diff
@@ -0,0 +1,397 @@
+			}
+
+		} else {
+			startID, err = getPkgID(ctx, gqlClient, opts.startPurl)
+
+			if err != nil {
+				logger.Errorf("error getting start pkg from purl inputted %s \n", err)
+				os.Exit(1)
+			}
+
+			if opts.stopPurl != "" {
+				stopPkg, err := getPkgID(ctx, gqlClient, opts.stopPurl)
+
+				if err != nil {
+					logger.Errorf("error getting stop pkg from purl inputted %s\n", err)
+				}
+
+				stopID = &stopPkg
+			}
+
+		}
+
+		bfsMap, path, err := analysis.SearchDependenciesFromStartNode(ctx, gqlClient, startID, stopID, opts.depth)
+
+		if err != nil {
+			logger.Errorf("error searching dependencies-- %s\n", err)
+			os.Exit(1)
+		}
+
+		frontiers, infoNodes, err := analysis.ToposortFromBfsNodeMap(ctx, gqlClient, bfsMap)
+
+		if err != nil {
+			logger.Errorf("error toposorting-- %s\n", err)
+			os.Exit(1)
+		}
+
+		poc := []string{}
+		for level := 0; level < len(frontiers); level++ {
+			frontierList := frontiers[level]
+			allNodes := []string{}
+			for _, id := range frontierList {
+				path = append(path, id)
+				allNodes = append(allNodes, id)
+			}
+			fmt.Printf("\n---FRONTIER LEVEL %d---\n", level)
+			poc = append(poc, printNodesInfo(ctx, gqlClient, bfsMap, allNodes)...)
+		}
+
+		fmt.Printf("\n---INFO NODES---\n")
+		if len(infoNodes) == 0 {
+			fmt.Printf("no info nodes found\n")
+		} else {
+			poc = append(poc, printNodesInfo(ctx, gqlClient, bfsMap, infoNodes)...)
+		}
+
+		fmt.Printf("\n---POINTS OF CONTACT---")
+		if len(poc) == 0 {
+			fmt.Printf("\nno POCs found\n")
+		} else {
+			for _, id := range poc {
+				fmt.Printf("\n%s: %s", id, makePOCPretty(bfsMap[id].PointOfContact))
+			}
+		}
+
+		fmt.Printf("\n\n---SUBGRAPH VISUALIZER URL--- \nhttp://localhost:3000/?path=%s\n", strings.Join(removeDuplicateValuesFromPath(path), `,`))
+	},
+}
+
+func printNodesInfo(ctx context.Context, gqlClient graphql.Client, bfsMap map[string]analysis.BfsNode, nodes []string) []string {
+	poc := []string{}
+	for _, id := range nodes {
+		node, err := model.Node(ctx, gqlClient, id)
+
+		if err != nil {
+			fmt.Printf("error: %s \n", err)
+		}
+
+		var pretty string
+		switch node := node.Node.(type) {
+		case *model.NodeNodePackage:
+			if bfsMap[id].Type == analysis.PackageName {
+				pretty = makePkgPretty(*node, false)
+			} else {
+				pretty = makePkgPretty(*node, true)
+			}
+		case *model.NodeNodeSource:
+			pretty = makeSrcPretty(*node)
+		case *model.NodeNodeArtifact:
+			pretty = makeArtifactPretty(*node)
+		}
+
+		fmt.Printf("%s: %s\n", id, pretty)
+
+		if bfsMap[id].PointOfContact.Email != "" {
+			poc = append(poc, id)
+		}
+	}
+	return poc
+}
+
+func makePOCPretty(poc model.AllPointOfContact) string {
+	return fmt.Sprintf("id- %s | email- %s | info- %s | collector- %s | justification- %s | origin- %s | since- %s", poc.Id, poc.Email, poc.Info, poc.Collector, poc.Justification, poc.Origin, poc.Since)
+}
+
+func makePkgPretty(pkg model.NodeNodePackage, version bool) string {
```

Comment:
can you use `PkgToPurl` here: https://github.com/guacsec/guac/blob/main/pkg/assembler/helpers/purl.go#L74